src/context/CartContext.jsx

```diff
@@ -0,0 +1,14 @@
+import { createContext, useContext, useState } from "react/cjs/react.development";
+
+
+const CartContext=createContext();
+
+export function CartProvider({children}){
+    const [cart, setCart]=useState([]);
+    return <CartContext.Provider value={{cart,setCart}}>
+        {children}
+    </CartContext.Provider> 
```

Comment:
Suggestion : Keep the Component designing part and API Logic different. Use context to access those functions on the basis of logic written in components